src/express.ts

```diff
@@ -16,6 +16,12 @@ import type {
 } from "./types";
 import { version } from "./version";
 
+/**
+ * lastRegisteredHash stores a checksum of the last successful function hash
+ * that was registered via this instance of the server.
+ */
+let lastRegisteredHash: string | undefined = undefined;
```

Comment:
Placing this here assumes we'll only ever have a single serve handler whereas it's technically possible to have multiple; it might be better to make this a property of the class instead.